crates/typst/src/engine.rs

```diff
@@ -202,6 +327,61 @@ impl Sink {
     }
 }
 
+/// Checks if a given warning is suppressed given one span it has a tracepoint
+/// in. If one of the ancestors of the node where the warning occurred has a
+/// warning suppression decorator sibling right before it suppressing this
+/// particular warning, the warning is considered suppressed.
+fn check_warning_suppressed(
+    span: Span,
+    world: &dyn World,
+    identifier: &diag::Identifier,
+) -> bool {
+    let Some(file) = span.id() else {
+        // Don't suppress detached warnings.
+        return false;
+    };
+
+    // The source must exist if a warning occurred in the file,
+    // or has a tracepoint in the file.
+    let source = world.source(file).unwrap();
```

Comment:
better safe than sorry imo, I would just skip it. could also combine it with the previous thing:

```rs
let Some(source) = span.id().and_then(|id| world.source(id)) else { return false };
```